src/coreclr/jit/optimizer.cpp

```diff
@@ -6637,49 +6634,39 @@ void Compiler::optHoistThisLoop(unsigned lnum, LoopHoistContext* hoistCtxt, Basi
         //
         // TODO-CQ: In future, we should create preheaders upfront before building
         // dominators so we don't have to do this extra work here.
-        BasicBlock*     cur            = pLoopDsc->lpExit;
-        BasicBlockList* preHeadersList = existingPreHeaders;
 
+        BasicBlock* cur = pLoopDsc->lpExit;
         while (cur != nullptr && pLoopDsc->lpContains(cur) && cur != pLoopDsc->lpEntry)
         {
             JITDUMP("  --  " FMT_BB " (dominate exit block)\n", cur->bbNum);
             defExec.Push(cur);
             cur = cur->bbIDom;
 
-            if (preHeadersList != nullptr)
+            for (; preHeadersList != nullptr; preHeadersList = preHeadersList->next)
```

Comment:
Did you find out an example where we didn't add a preheader in this loop?

Reply:
> Did you find out an example where we didn't add a preheader in this loop?

Do you mean where we added more than one preheader? Zero or one would happen the same as before. (And the answer is yes)

Reply:
Ok and the example is https://github.com/dotnet/runtime/pull/71504#issuecomment-1171638776?

Reply:
Correct.